### PR TITLE
improve compiler api

### PR DESCRIPTION
**Pull request by yann300**
fix https://github.com/ethereum/remix-ide/issues/2712
partly_fix https://github.com/ethereum/remix-ide/issues/2854

 - make run module to listen on yulp plugin
 - add `compileWithParameters` and `setCompilerConfig` plugin API

### CODE REVIEW

File: src/app/compiler/compiler-artefacts.js [hunk 1]
```diff
@@ -32,6 +32,10 @@ module.exports = class CompilerArtefacts extends Plugin {
     this.on('lexon', 'compilationFinished', (file, source, languageVersion, data) => {
       this.compilersArtefacts['__last'] = new CompilerAbstract(languageVersion, data, source)
     })
+
+    this.on('yupl', 'compilationFinished', (file, source, languageVersion, data) => {
```
Comment (LianaHus) [inline src/app/compiler/compiler-artefacts.js:36]: is this a typo?

File: src/app/tabs/compile-tab.js [hunk 2]
```diff
@@ -217,6 +238,31 @@ class CompileTab extends ViewPlugin {
...
+   * This function is used by remix-plugin compiler API.
+   * @param {object} settings {evmVersion, optimize, compilerUrl, version, language}
+   */
+  setCompilerConfig (settings) {
+    return new Promise((resolve, reject) => {
+      addTooltip(yo`<div><b>${this.currentRequest.from}</b> is updating the <b>Solidity compiler configuration</b>.<pre class="text-left">${JSON.stringify(settings, null, '\t')}</pre></div>`)
+      this.compilerContainer.setLanguage(settings.language)
+      this.compilerContainer.setEvmVersion(settings.evmVersion)
+      this.compilerContainer.setOptimize(settings.optimize)
+      this.compilerContainer.setVersion(settings.version)
```
Comment (LianaHus) [inline src/app/tabs/compile-tab.js:252]: sould maybe compilerContainer has setSettings?

File: src/app/compiler/compiler-artefacts.js (hunk 1 above)
Comment (yann300) [reply src/app/compiler/compiler-artefacts.js:36]: indeed yes it is https://github.com/ethereum/remix-plugins-directory/blob/master/plugins/yulp/profile.json

File: src/app/tabs/compile-tab.js (hunk 2 above)
Comment (yann300) [reply src/app/tabs/compile-tab.js:252]: yes sure, the thing is that these methods were already existing before

**Pull request merged**